src/lib/orionld/serviceRoutines/orionldPostEntityOperationsUpsert.cpp

```diff
@@ -0,0 +1,331 @@
+
+    char  typeExpanded[256];
+    char* details;
+
+    entityIdP->id = entityId;
+    if (orionldUriExpand(orionldState.contextP, entityType, typeExpanded, sizeof(typeExpanded), &details) == false)
+    {
+      orionldErrorResponseCreate(OrionldBadRequestData, "Error during URI expansion of entity type", details, OrionldDetailString);
+      return false;
+    }
+    entityIdP->type      = typeExpanded;
+    entityIdP->isPattern = "false";
+    entityIdP->creDate   = getCurrentTime();
+    entityIdP->modDate   = getCurrentTime();
+
+    if (kjTreeToContextElementAttributes(ciP, entityNodeP->value.firstChildP, createdAtP, modifiedAtP, ceP) == false)
+    {
+      // kjTreeToContextElement fills in error using 'orionldErrorResponseCreate()'
+      mongoRequest.release();
+      LM_E(("kjTreeToContextElement failed"));
+      return false;
+    }
+    orionldState.payloadIdNode = NULL;
+    orionldState.payloadTypeNode = NULL;
+    ++ix;
+  }
+  // }
+
+  //
+  // Call mongoBackend
+  //
+  ciP->httpStatusCode = mongoUpdateContext(&mongoRequest,
+                                           &mongoResponse,
+                                           orionldState.tenant,
+                                           ciP->servicePathV,
+                                           ciP->uriParam,
+                                           ciP->httpHeaders.xauthToken,
+                                           ciP->httpHeaders.correlator,
+                                           ciP->httpHeaders.ngsiv2AttrsFormat,
+                                           ciP->apiVersion,
+                                           NGSIV2_NO_FLAVOUR);
+
+  //
+  // Now check orionldState.errorAttributeArray to see whether any attribute failed to be updated
+  //
+  // bool partialUpdate = (orionldState.errorAttributeArrayP[0] == 0)? false : true;
+  // bool retValue      = true;
+
+  LM_E(("mongoUpdateContext: HTTP Status Code: %d", ciP->httpStatusCode));
+
+  if (ciP->httpStatusCode == SccOk)
+  {
+    LM_TMP(("Items in mongoResponse.contextElementResponseVector: %d", mongoResponse.contextElementResponseVector.size()));
+    orionldState.responseTree = kjObject(orionldState.kjsonP, NULL);
+
+    KjNode* successArrayP = kjArray(orionldState.kjsonP, "success");
+
+    for (unsigned int ix = 0; ix < mongoResponse.contextElementResponseVector.vec.size(); ix++)
+    {
+      KjNode* nodeP = kjString(orionldState.kjsonP, "id", mongoResponse.contextElementResponseVector.vec[ix]->contextElement.entityId.id.c_str());
+
+      //
+      // FIXME: Here we assume all items in mongoResponse.contextElementResponseVector are SUCCESSFUL
+      //        But, ContextElementResponse has a field called "statusCode" that we could look at ...
+      //        The difficult part is ... how to test this?
+      //        How provoke a failure?
+      //        This service Updates if entity exists and creates if not.
+      //        Very difficult to provoke an error ... :(
+      //
```

Comment:
This is the comment I was talking about - look at statusCode and see if its `code` is 200 or something else.

As we said, easily provoked by including an entity with invalid Entity ID, type missing, type/id not a JSON string, etc, etc

Reply:
FIXED